core/dbt/logger.py

```diff
@@ -360,8 +361,8 @@ def process(self, record):
 
 
 def make_log_dir_if_missing(log_dir):
-    import dbt.clients.system
-    dbt.clients.system.make_directory(log_dir)
+    # N.B:  Storage adapters can't be used in the logger (circular imports)
```

Comment:
Oh this is an interesting point. I guess we can always write logs locally and then if they need to be stored somewhere else for cloud, we can move them

Reply:
might want to add "syncing logs" as something to think about when we work on structured logging.

Reply:
Alternatively, we should consider the storage adapter taking over logging. It might be a bit involved, but changing all the log calls to be SA log calls would really only involve finagling import statements. I don't particularly like logs going somewhere else outside the configured SA.

Reply:
I agree that storage adapters need to be used for logging (somehow) for the long term. Probably not necessary for this stage though.  Another issue is that I'm pretty sure that we will need to put different files in different places eventually. For example the logs, json artifacts, partial parsing file, compiled SQL, all have different use cases that will probably require different storage destinations.

Reply:
Having that all centralized will help manage all the different destinations. If we don't merge logging into SAs we should definitely make a ticket to track it for when we're ready to.

Reply:
+1 to all of this.. right now we have a chicken/egg problem in that we use the logger in the SA to log requests so we can't import the logger which is importing the SA, which is importing the logger... etc...

We can certainly re-work how that goes, but not for this effort.